tools/test/test262/test-data-runner.cpp

```diff
@@ -0,0 +1,137 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <atomic>
+#include <vector>
+#include <thread>
+#include <algorithm>
+
+struct TestData {
+    std::string canBlockIsFalse;
+    std::string isModule;
+    std::string fullPath;
+    std::string driverFile;
+    std::string testFile;
+    std::string code;
+};
+
+std::string g_inputPath;
+std::vector<TestData> g_testDatas;
+std::atomic<int> g_index;
+std::atomic<int> g_passCount;
+std::atomic<int> g_skipCount;
+bool g_enableIntl = true;
+
+int main(int argc, char* argv[])
+{
+    std::string shellPath = "escargot";
+    if (argc >= 2) {
+        shellPath = argv[1];
+    }
+    int numThread = std::thread::hardware_concurrency();
+    if (argc >= 3) {
+        numThread = std::stoi(argv[2]);
+    }
+    if (argc >= 4) {
+        g_enableIntl = false;
+    }
+
+    g_inputPath = "test262_data";
+    std::ifstream input(g_inputPath + "/data");
+
+    std::string canBlockIsFalse;
+    std::string isModule;
+    std::string fullPath;
+    std::string driverFile;
+    std::string testFile;
+    std::string code;
+
+    int caseNum = 0;
+    while (std::getline(input, canBlockIsFalse)) {
+        std::getline(input, isModule);
+        std::getline(input, fullPath);
+        std::getline(input, driverFile);
+        std::getline(input, testFile);
+        std::getline(input, code);
+
+        g_testDatas.push_back({
+            canBlockIsFalse,
+            isModule,
+            fullPath,
+            driverFile,
+            testFile,
+            code
+        });
+        caseNum++;
+    }
+
+    printf("Total case number %d\n", caseNum);
+
+    std::vector<std::pair<int, int>> threadData;
+    std::vector<std::thread> threads;
+    int threadSize = caseNum / numThread;
+
+    int d = 0;
+    for (int i = 0; i < numThread; i ++) {
+        threadData.push_back(std::make_pair(
+            d, d + threadSize
+            ));
+        d += threadSize;
+    }
+
+    threadData.back().second = caseNum;
+
+    for (int i = 0; i < numThread; i ++) {
+        printf("%d -> %d\n", threadData[i].first, threadData[i].second);
+        threads.push_back(std::thread([](std::pair<int, int> data, std::string shellPath) {
+            for (int j = data.first; j < data.second; j ++) {
+                std::string commandline = "TZ=US/Pacific " + shellPath;
+                const auto& data = g_testDatas[j];
+
+                if ((data.fullPath.find("/intl402/") != std::string::npos ||
+                    data.fullPath.find("/Temporal/") != std::string::npos) && !g_enableIntl) {
```

Comment:
Why intl* and Temporal skipped?
For the fast scan of test262 running?

Reply:
not every intl tests could run on android due to ICU version on android(or devices)

Reply:
I see